test-poc/framework/src/main/java/org/keycloak/test/framework/injection/Registry.java

```diff
@@ -209,10 +211,47 @@ private InstanceWrapper getDeployedInstance(Supplier supplier) {
     }
 
     private void loadSuppliers() {
-        ServiceLoader.load(Supplier.class).iterator().forEachRemaining(suppliers::add);
+        Iterator<Supplier> supplierIterator = ServiceLoader.load(Supplier.class).iterator();
+        Set<Class> loadedValueTypes = new HashSet<>();
+        Set<Supplier> skippedSuppliers = new HashSet<>();
+
+        while (supplierIterator.hasNext()) {
+            Supplier supplier = supplierIterator.next();
+            boolean shouldAdd = false;
+
+            if (!loadedValueTypes.contains(supplier.getValueType())) {
+                String requestedSupplier = Config.getInstance().getSelectedSupplier(supplier.getValueType());
+                if (requestedSupplier != null) {
+                    if (requestedSupplier.equals(supplier.getAlias())) {
+                        shouldAdd = true;
+                    }
+                } else {
+                    shouldAdd = !loadedValueTypes.contains(supplier.getValueType());
```

Comment:
This is always `true`, as it matches the condition of the `if` it is contained in.
```suggestion
                    shouldAdd = true;
```

Reply:
Updated